src/bundle/Core/Command/Indexer/ContentIdList/ContentTypeInputGeneratorStrategy.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Bundle\Core\Command\Indexer\ContentIdList;
+
+use Generator;
+use Ibexa\Bundle\Core\Command\Indexer\ContentIdListGeneratorStrategyInterface;
+use Ibexa\Contracts\Core\Repository\ContentService;
+use Ibexa\Contracts\Core\Repository\Values\Content\ContentList;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query;
+use Ibexa\Contracts\Core\Repository\Values\Filter\Filter;
+use Symfony\Component\Console\Input\InputInterface;
+
+/**
+ * @internal
+ */
+final class ContentTypeInputGeneratorStrategy implements ContentIdListGeneratorStrategyInterface
+{
+    private ContentService $contentService;
+
+    /** @var array<string, \Ibexa\Contracts\Core\Repository\Values\Content\ContentList> */
+    private static array $inMemoryContentListCache = [];
+
+    public function __construct(ContentService $contentService)
+    {
+        $this->contentService = $contentService;
+    }
+
+    /**
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\BadStateException
+     */
+    public function getGenerator(InputInterface $input, int $iterationCount): Generator
+    {
+        $contentList = $this->getContentList($input->getOption('content-type'));
+        $contentListIterator = $contentList->getIterator();
+        do {
+            $contentIds = [];
+            for ($i = 0; $i < $iterationCount; ++$i) {
+                if (!$contentListIterator->valid()) {
+                    break;
+                }
+                $contentIds[] = $contentListIterator->current()->getVersionInfo()->getContentInfo()->getId();
+                $contentListIterator->next();
+            }
+            if (empty($contentIds)) {
+                break;
+            }
+
+            yield $contentIds;
+        } while ($contentListIterator->valid());
```

Comment:
It's a weird mix of `do...while` loop with `for` (it should also be a `while + for` loop if anything).

I'd suggest using foreach instead, since it will basically execute the very same logic as you did, but not relying on it being an iterator directly.

```suggestion
        $contentIds = [];
        foreach ($contentList as $content) {
            $contentIds[] = $contentListIterator->current()->getVersionInfo()->getContentInfo()->getId();

            if (count($contentIds) >= 5) {
                yield $contentIds;
                $contentIds = [];
            }
        }

        if (!empty($contentIds)) {
            yield $contentIds;
        }
```

You can use `count()` here, or just increment a variable. Either is fine, as the operation is too quick to make a difference (in comparison to everything else that might be happening).

Reply:
Why
```php
if (count($contentIds) >= 5) {
```
?

Reply:
Fixed via d95a37b7c1 - this is probably how it was supposed to look.

Reply:
> Why
> 
> ```
> if (count($contentIds) >= 5) {
> ```
> 
> ?

As said, you can either execute `count()` (which uses array reference counter, so it's fast) or just keep track through a variable. It's just detail.